addons/survey/controllers/main.py

```diff
@@ -246,27 +227,31 @@ def submit(self, survey, **post):
             ret['errors'] = errors
         else:
             # Store answers into database
-            user_input_obj = request.registry['survey.user_input']
+            Input = request.env['survey.user_input']
 
-            user_input_line_obj = request.registry['survey.user_input_line']
+            InputLine = request.env['survey.user_input_line']
             try:
-                user_input_id = user_input_obj.search(cr, SUPERUSER_ID, [('token', '=', post['token'])], context=context)[0]
+                user_input = Input.sudo().search([('token', '=', post['token'])], limit=1)
             except KeyError:  # Invalid token
                 return request.website.render("website.403")
```

Comment:
probably because the old api code had something like [0] (line 253) and now we simply have to check that user_input is empty instead of doing a try/except